src/testing/alerts.rs

```diff
@@ -0,0 +1,525 @@
+    let alert_hash = Signable::hash(&alert);
+    let signed_alert = test_case
+        .unchecked_signed(alert.clone(), alerter_index)
+        .await;
+    test_case
+        .incoming_message(AlertMessage::ForkAlert(signed_alert))
+        .outgoing_notification(ForkingNotification::Forker(fork_proof))
+        .wait();
+    for i in 1..n_members.0 - 1 {
+        let node_id = NodeIndex(i);
+        let signed_alert_hash = test_case
+            .indexed_unchecked_signed(alert_hash, node_id)
+            .await;
+        test_case.incoming_message(AlertMessage::RmcMessage(
+            node_id,
+            RmcMessage::SignedHash(signed_alert_hash),
+        ));
+    }
+    test_case.outgoing_notification(ForkingNotification::Units(Vec::new()));
+    test_case.run(own_index).await;
+}
+
+#[tokio::test]
+async fn asks_about_unknown_alert() {
+    let n_members = NodeCount(7);
+    let own_index = NodeIndex(0);
+    let alerter_index = NodeIndex(1);
+    let forker = NodeIndex(6);
+    let mut test_case = TestCase::new(n_members);
+    let fork_proof = test_case.fork_proof(forker, 0).await;
+    let alert = test_case.alert(alerter_index, fork_proof.clone());
+    let alert_hash = Signable::hash(&alert);
+    let signed_alert_hash = test_case
+        .indexed_unchecked_signed(alert_hash, alerter_index)
+        .await;
+    test_case
+        .incoming_message(AlertMessage::RmcMessage(
+            alerter_index,
+            RmcMessage::SignedHash(signed_alert_hash),
+        ))
+        .outgoing_message(
+            AlertMessage::AlertRequest(own_index, alert_hash),
+            Recipient::Node(alerter_index),
+        );
+    test_case.run(own_index).await;
+}
+
+#[tokio::test]
+async fn ignores_wrong_alert() {
+    let n_members = NodeCount(7);
+    let own_index = NodeIndex(0);
+    let alerter_index = NodeIndex(1);
+    let forker = NodeIndex(6);
+    let mut test_case = TestCase::new(n_members);
+    let valid_unit = test_case.unchecked_signed_unit(alerter_index, 0, 0).await;
+    let wrong_fork_proof = (valid_unit.clone(), valid_unit);
+    let wrong_alert = test_case.alert(forker, wrong_fork_proof.clone());
+    let signed_wrong_alert = test_case
+        .unchecked_signed(wrong_alert.clone(), forker)
+        .await;
+    let signed_wrong_alert_hash = test_case
+        .indexed_unchecked_signed(Signable::hash(&wrong_alert), own_index)
+        .await;
+    test_case
+        .incoming_message(AlertMessage::ForkAlert(signed_wrong_alert))
+        .unexpected_notification(ForkingNotification::Forker(wrong_fork_proof));
+    for i in 1..n_members.0 {
+        test_case.unexpected_message(
+            AlertMessage::RmcMessage(
+                own_index,
+                RmcMessage::SignedHash(signed_wrong_alert_hash.clone()),
+            ),
+            Recipient::Node(NodeIndex(i)),
+        );
+    }
+    // We also make a proper alert to actually have something to wait for.
```

Comment:
What do you mean?

Reply:
`TestCase` only waits for messages while it has any messages it expects. If we don't expect any message it will just finish successfully immediately.